packages/components/notification-center/notification-center.service.ts

```diff
@@ -0,0 +1,167 @@
+import { EventEmitter, inject, Injectable, TemplateRef } from '@angular/core';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { DateAdapter, DateFormatter } from '@koobiq/components/core';
+import { KbqToastData, KbqToastService, KbqToastStyle } from '@koobiq/components/toast';
+import { BehaviorSubject, combineLatestWith, merge, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+export interface KbqNotificationItem extends Omit<KbqToastData, 'closeButton'> {
+    id?: string;
+
+    title?: string | TemplateRef<unknown>;
+    style?: string | KbqToastStyle;
+
+    icon?: boolean | TemplateRef<unknown>;
+    iconClass?: string;
+    caption?: string | TemplateRef<unknown>;
+
+    content?: string | TemplateRef<unknown>;
+    actions?: TemplateRef<unknown>;
+
+    date: string;
+    read?: boolean;
+}
+
+type KbqNotificationsGroup = { title: string; items: KbqNotificationItem[] };
+
+type KbqNotificationsGroups = Record<string, KbqNotificationsGroup>;
+
+@Injectable({ providedIn: 'root' })
+export class KbqNotificationCenterService<D> {
+    /** @docs-private */
+    private readonly adapter = inject(DateAdapter<D>);
+    /** @docs-private */
+    private readonly formatter = inject(DateFormatter<D>);
+    /** @docs-private */
+    private readonly toastService = inject(KbqToastService);
+
+    /** @docs-private */
+    readonly silentMode = new BehaviorSubject(false);
+    /** @docs-private */
+    readonly loadingMode = new BehaviorSubject(false);
+    /** @docs-private */
+    readonly errorMode = new BehaviorSubject(false);
+    /** @docs-private */
+    readonly onRead = new BehaviorSubject<KbqNotificationItem | null>(null);
+
+    /** Triggers an event when the user presses the reload button. */
+    readonly onReload = new EventEmitter<void>();
+
+    private originalItems = new BehaviorSubject([] as KbqNotificationItem[]);
+
+    /** @docs-private */
+    readonly groupedItems = this.originalItems.pipe(
+        map((items) => {
+            const result: KbqNotificationsGroups = {};
+
+            items.map((item) => this.makeGroup(item, result));
+
+            return Object.values(result).reverse();
+        })
+    );
+
+    /** Emits an event whenever the changes. */
+    readonly changes = merge(this.silentMode, this.loadingMode, this.errorMode, this.originalItems, this.onRead);
+
+    /** Notification items */
+    get items() {
+        return this.originalItems.value;
+    }
+
+    set items(values: KbqNotificationItem[]) {
+        this.originalItems.next(this.setReadState(this.setIds(values)));
+    }
+
+    /** Number of unread notifications */
+    get unreadItems(): Observable<number> {
```

Comment:
Предлагаю использовать `Length` в названии во избежание путаницы т.к. есть `originalItems` и `groupedItems` , которые н есут информацию об объектах, а тут количество

```suggestion
    get unreadItemsLength(): Observable<number> {
```

Reply:
`Counter` обычно + тут нужно обработать вариант с `99+`, поэтому настоящей длины тут не будет